### PR TITLE
transformer implementation

### PR DESCRIPTION
**Pull request by megakorre**
here is a implementation of transformers to compare.

DISCUSS!!!! :smile:

### CODE REVIEW

File: src/moar/core.clj [hunk 1]
```diff
@@ -26,10 +26,8 @@
 (defn bind
   "Applies a function returning a monad to a monad of the same kind"
   ([monad function]
-     {:pre [(satisfies? MonadInstance monad)]}
      (bind (->monad-implementation monad) monad function))
   ([impl monad function]
-     {:post [(same-monad? monad %)]}
```
Comment (Jell) [inline src/moar/core.clj:32]: Why was this removed? It was very useful for me when implementing stuff

Comment (megakorre) [inline src/moar/core.clj:32]: it can be added back if equality is implemented in all monadTransformerImplementation instances.
witch maybe is useful for development.
Maybe theres a smarter way to do it. that test is pretty useful in testing

File: src/moar/monads/maybe.clj [hunk 2]
```diff
@@ -12,17 +15,53 @@
...
+
   MonadPlus
   (mzero* [_] nothing)
   (mplus* [_ monad-a monad-b]
-    (case [(just? monad-a) (just? monad-b)]
-      [false false] nothing
-      [ true false] monad-a
-      [false  true] monad-b
-      [ true  true] monad-a)))
+    (maybe-or (monad-a monad-b))))
```
Comment (Jell) [inline src/moar/monads/maybe.clj:26]: that's probably broken? :) should read `(maybe-or monad-a monad-b)`

**Comment by Jell** [comment]
Yo! I did my own implementation of lift: https://github.com/Jell/moar/blob/master/test/moar/core_test.clj

Let's take the time to compare the two and discuss about it when we get the time. Cheers!

**Comment by megakorre** [comment]
Yea well have to discuss it your version of lift is not the one in my implementation

http://www.haskell.org/haskellwiki/Lifting 

yours can lift pure functions but whats more interesting is to be able to lift monadic values from the different monads into the transformer.

https://github.com/Jell/moar/pull/5/files#diff-e4167bd2bc43200a63c7e03fd2304e4fR34

**Comment by Jell** [comment]
I've change so that my version can lift pure values, monadic values, pure functions and monadic functions.
Is it closer to yours or am I missing the point?
https://github.com/Jell/moar/blob/master/test/moar/core_test.clj#L44-L52

**Comment by Jell** [comment]
From what I see, the biggest difference is the "lower" method that requires a "wrapping monad" for each non-transformer monad, and the "transform-impl" that takes a vector of monads (also requiring a wrapping monad).
So I guess each monad should know it's own transformer like in your implementation, but perhaps lift could be left general like in mine?
Is there a case where lift might need to be specialised in a non-general way?
